tests/functional/cli/test_cli_variables.py

```diff
@@ -17,3 +26,59 @@ def test_list_project_variables(gitlab_cli, project):
     ret = gitlab_cli(cmd)
 
     assert ret.success
+
+
+def test_list_project_variables_with_path(gitlab_cli, project):
+    cmd = ["project-variable", "list", "--project-id", project.path_with_namespace]
+    ret = gitlab_cli(cmd)
+
+    assert ret.success
+
+
+@pytest.fixture
+def resp_get_project():
+    return {
+        "method": responses.GET,
+        "url": f"{DEFAULT_URL}/api/v4/projects/1",
+        "json": {"name": "name", "path": "test-path", "id": 1},
+        "content_type": "application/json",
+        "status": 200,
+    }
+
+
+@pytest.mark.script_launch_mode("inprocess")
+@responses.activate
+def test_list_project_variables_with_path_url_check(
+    monkeypatch, script_runner, resp_get_project
+):
+    monkeypatch.setattr(config, "_DEFAULT_FILES", [])
+    resp_get_project_in_ci = copy.deepcopy(resp_get_project)
+    resp_get_project_in_ci.update(
+        url=f"{DEFAULT_URL}/api/v4/projects/project%2Fwith%2Fa%2Fnamespace/variables"
+    )
+    resp_get_project_in_ci.update(
+        json=[
+            {
+                "variable_type": "env_var",
+                "key": "TEST_VARIABLE_1",
+                "value": "TEST_1",
+                "protected": False,
+                "masked": True,
+                "environment_scope": "*",
+            },
+            {
+                "variable_type": "env_var",
+                "key": "TEST_VARIABLE_2",
+                "value": "TEST_2",
+                "protected": False,
+                "masked": False,
+                "environment_scope": "*",
+            },
+        ]
+    )
```

Comment:
I think that we might not even need to update this, since we only care about matching the URL being sent, not what we do with the data returned, right?

Reply:
Well we need to make sure we return a list. I will make it an empty list.

Reply:
ahh ok sure. ok, yeah that or just a subset of items is probably enough to keep the boilerplate down. Thanks!